src/jit/morph.cpp

```diff
@@ -4491,6 +4491,19 @@ void                Compiler::fgMoveOpsLeft(GenTreePtr tree)
             new_op1->gtType = TYP_I_IMPL;
         }
 
+        // If new_op1 is a new expression. Assign it a new unique value number.
+        if (new_op1->gtVNPair.GetLiberal() != ValueNumStore::NoVN)
```

Comment:
Instead of checking for NoVN we normally check to see if Value Numberring has been run yet, this way:

                        // vnStore is null before the ValueNumber phase has run
                        if (vnStore != nullptr)